rework attribute dict

pSpitzner opened the pull request:
we likely do not need our attribute dict implementation anymore, a simple dict should be sufficient.

this should also make it easier to deal with aliases,
and fix the issue of the `ConfigExtra` where aliased fields do not match between schema data and extra data.

pSpitzner committed:
added a marker where we need to look at

pSpitzner committed:
removed attribute dict in favor for simple dictionary.
we loose attribute access to nested and non-existing fields.

pSpitzner committed:
ruff and moved one test

pSpitzner committed:
renamed helper attributes in access proxy with underscorde. we should consider
doning the same for `to_dict()` to avoid problems with a potential "to_dict" field.

semohr commented on src/eyconf/config/base.py:
```diff
@@ -100,48 +101,58 @@ def _update(
                 if "alias" in f.metadata
             }
 
```
I don’t think `attr_key` and `dict_key` are good names here. The base class shouldn’t need to care about these distinctions, it only cares about aliases. Maybe we can use `alias_key` here instead?

semohr committed:
Added a test for how aliases are handled in the json schema generator.

semohr committed:
Instead of passing the dict and attr keys we can use the dataclass fields
directly. This allows us to simplify the update function quite a bit.

This should remove the necessity to add the `resolve_alias_*` functions.

semohr committed:
Added a function to unify metadata lookup from dataclasses.
Removed resolve_alias_* functions.

semohr committed:
Ruff lint

pSpitzner committed:
minor fixes and renames around access proxy with @semohr

semohr committed:
Renamed attr_key to key. Fixed some typing issues

semohr committed:
Fixed some minor naming inconsistencies

semohr committed:
Added changelog entry and bumped version

pSpitzner committed:
Minor renames
